### PR TITLE
Making batching of usage configurable

### PR DESCRIPTION
**Pull request by codatoz**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ func main() {
 					{Name: snowflake.SfDataUsageWindow, Description: "The maximum number of days of usage data to retrieve. Default is 90. Maximum is 90 days.", Mandatory: false},
 					{Name: snowflake.SfDatabaseRoles, Description: "If set, database-roles for all databases will be fetched.", Mandatory: false},
 					{Name: snowflake.SfIgnoreLinksToRoles, Description: "This comma separated list of regular expressions can be used to indicate that role hierarchy links to certain roles are never added or removed. e.g. 'SYS.+,ADMIN.+' will match all roles starting with 'SYS' or 'ADMIN', meaning that all grants to these roles will remain untouched during the sync.", Mandatory: false},
+					{Name: snowflake.SfUsageBatchSize, Description: "If not set, no batching is done for fetching usage. This will be the fastest but usage more memory. If memory usage is a problem, this can be set to a number between 10.000 and 1.000.000 (higher is recommended) to fetch usage in batches of that size.", Mandatory: false},
```
Comment (TimmersThomas) [inline main.go:50]: `but usage more memory` <-- typo?

File: snowflake/data_usage.go [hunk 2]
```diff
@@ -107,14 +108,32 @@ func (s *DataUsageSyncer) SyncDataUsage(ctx context.Context, fileCreator wrapper
...
+
 	return nil
 }
 
+func logUsageBatch(count int) {
+	var m runtime.MemStats
+	runtime.ReadMemStats(&m)
+
+	// The following outputs are in bytes
+	fmt.Printf("\tHeap = %v MiB", bToMb(m.Alloc))
```
Comment (TimmersThomas) [inline snowflake/data_usage.go:126]: specific reason for not using `logger.` ?

File: snowflake/repository.go [hunk 3]
```diff
@@ -60,9 +62,22 @@ func NewSnowflakeRepository(params map[string]string, role string) (*SnowflakeRe
...
 
+	usageBatchSize := 0
+	if v, f := params[SfUsageBatchSize]; f && v != "" {
+		usageBatchSize, err = strconv.Atoi(v)
+		if err != nil {
+			return nil, fmt.Errorf("parsing %q parameter: %w", SfUsageBatchSize, err)
+		}
+
+		if usageBatchSize != 0 && (usageBatchSize < 10000 || usageBatchSize > 1000000) {
+			return nil, fmt.Errorf("invalid value %d for %q parameter (If set, it must not be smaller than 10.000 and not larger than 1.000.000)", usageBatchSize, SfUsageBatchSize)
```
Comment (TimmersThomas) [inline snowflake/repository.go:73]: `it must be between 10.000 and 1.000.00` ?
Reads a bit easier imho

File: snowflake/repository.go [hunk 4]
```diff
@@ -82,9 +97,7 @@ func (repo *SnowflakeRepository) isProtectedRoleName(rn string) bool {
 }
 
 func (repo *SnowflakeRepository) GetDataUsage(ctx context.Context, minTime time.Time, maxTime *time.Time) <-chan stream.MaybeError[UsageQueryResult] {
-	batchSize := 20000
-
-	outputChannel := make(chan stream.MaybeError[UsageQueryResult], batchSize)
+	outputChannel := make(chan stream.MaybeError[UsageQueryResult], 10000)
```
Comment (TimmersThomas) [inline snowflake/repository.go:100]: Just wondering the `10.000` static value is no issue here?

**Review by TimmersThomas — DISMISSED** [review]

File: snowflake/repository.go (hunk 4 above)
Comment (codatoz) [reply snowflake/repository.go:100]: This is 'just' the buffer size for the channel. Writing will block if the buffer is full, but that's not a problem (+ will probably never happen).

**Review by TimmersThomas — APPROVED** [review]

**Pull request merged**